plugins/rule_engines/irods_rule_engine_plugin-passthrough/libirods_rule_engine_plugin-passthrough.cpp

```diff
@@ -0,0 +1,162 @@
+#include "irods_plugin_context.hpp"
+#include "irods_re_plugin.hpp"
+#include "irods_get_full_path_for_config_file.hpp"
+#include "irods_logger.hpp"
+#include "irods_state_table.h"
+#include "rodsError.h"
+
+#include <string>
+#include <functional>
+#include <vector>
+#include <regex>
+#include <fstream>
+
+#include "json.hpp"
+
+using pluggable_rule_engine = irods::pluggable_rule_engine<irods::default_re_ctx>;
+
+namespace {
+
+//
+// Rule Engine Plugin
+//
+
+using log = irods::experimental::log;
+
+struct pep_config
+{
+    std::regex regex;
+    int code;
+};
+
+std::vector<pep_config> pep_configs;
+int matched_code = 0;
+
+template <typename ...Args>
+using operation = std::function<irods::error(irods::default_re_ctx&, Args...)>;
+
+irods::error start(irods::default_re_ctx&, const std::string& _instance_name)
+{
+    std::string config_path;
+
+    if (auto error = irods::get_full_path_for_config_file("server_config.json", config_path);
+        !error.ok())
+    {
+        std::string msg = "server configuration not found [path => ";
+        msg += config_path;
+        msg += ']';
+
+        // clang-format off
+        log::rule_engine::error({{"rep", "passthrough"},
+                                 {"function", __func__},
+                                 {"msg", msg}});
+        // clang-format on
+
+        return ERROR(SYS_CONFIG_FILE_ERR, msg.c_str());
+    }
+
+    // clang-format off
+    log::rule_engine::trace({{"rep", "passthrough"},
+                             {"function", __func__},
+                             {"msg", "reading plugin configuration ..."}});
+    // clang-format on
+
+    nlohmann::json config;
+
+    {
+        std::ifstream config_file{config_path};
+        config_file >> config;
+    }
+
+    try {
+        for (const auto& re : config.at("plugin_configuration").at("rule_engines")) {
+            if (_instance_name == re.at("instance_name").get<std::string>()) {
+                for (const auto& e : re.at("plugin_specific_configuration").at("return_codes_for_peps")) {
+                    pep_configs.push_back({std::regex{e.at("regex").get<std::string>()}, e.at("code").get<int>()});
+                }
+
+                return SUCCESS();
+            }
+        }
```

Comment:
Would it be helpful to break out the contents of the try block into a separate function and/or leaving an explanatory comment? Seems pretty straightforward, but I had to stare at it for a second to understand what was happening.

Reply:
It should use the keywords if available.  I will update the code to use `irods::CFG_PLUGIN_SPECIFIC_CONFIGURATION_KW`.

I thought about making new ones, but other plugins do not define their keywords under `irods/irods`.  So, I think I'm going to leave the plugin specific keywords as is unless others feel differently.